components/TxForm.tsx

```diff
@@ -62,77 +38,20 @@ export const TxForm = () => {
   const [tokenAddress, setTokenAddress] = useState<string>();
   const [tokenAmount, setTokenAmount] = useState<string>("");
   const [tokenDecimals, setTokenDecimals] = useState<number>();
+  const { shield } = useRailgunTx();
 
   const doSubmit: React.FormEventHandler = async (e) => {
     // TODO: Form validation
-    // Formatted token amounts and recipients.
-    const erc20AmountRecipients: RailgunERC20AmountRecipient[] = [
-      {
-        tokenAddress: tokenAddress!,
-        amountString: ethers.utils
-          .parseUnits(tokenAmount!, tokenDecimals)
-          .toHexString(), // must be hex
-        recipientAddress: recipient!, // RAILGUN address
-      },
-    ];
-
-    // The shieldPrivateKey enables the sender to decrypt
-    // the receiver's address in the future.
-    const shieldPrivateKey = await getShieldPrivateKey();
-
-    if (!doErc20Approval) throw "not prepared";
-    await doErc20Approval();
-
-    // Public wallet to shield from.
-    const fromWalletAddress = address as `0x{string}`;
-
-    const { gasEstimateString, error: err } = await gasEstimateForShield(
-      network,
-      shieldPrivateKey,
-      erc20AmountRecipients,
-      [], // nftAmountRecipients
-      fromWalletAddress
-    );
-
-    if (err) {
-      throw err;
-    }
-
-    const { maxFeePerGas, maxPriorityFeePerGas } = await provider.getFeeData();
-    const gasDetailsSerialized: TransactionGasDetailsSerialized = {
-      evmGasType: EVMGasType.Type2, // Depends on the chain (BNB uses type 0)
-      gasEstimateString: gasEstimateString!, // Output from gasEstimateForShield
-      maxFeePerGasString: maxFeePerGas!.toHexString(), // Current gas Max Fee
-      maxPriorityFeePerGasString: maxPriorityFeePerGas!.toHexString(), // Current gas Max Priority Fee
-    };
-
-    const { serializedTransaction, error } = await populateShield(
-      network,
-      shieldPrivateKey,
-      erc20AmountRecipients,
-      [], // nftAmountRecipients
-      gasDetailsSerialized
-    );
-    if (error) {
-      throw error;
-    }
-
-    const { chain } = NETWORK_CONFIG[network];
-
-    const transactionRequest: ethers.providers.TransactionRequest =
-      deserializeTransaction(
-        serializedTransaction as string,
-        undefined, // nonce (optional)
-        chain.id
-      );
-
-    // Public wallet to shield from.
-    transactionRequest.from = address;
-    // TODO: handle transaction status and notifications in a feature
-    const tx = await signer?.sendTransaction(transactionRequest);
-    await tx?.wait().then(() => {
-      txNotify(tx.hash);
+    if (!tokenAddress || !tokenAmount || !tokenDecimals || !recipient)
```

Comment:
Is checking both `tokenDecimals` and `tokenAddress` redundant? Since they're both populated from the token dropdown

Reply:
perhaps redundant. the alternative is assume the token dropdown is fulfilling its duties.

and typescript won't let me!

<img width="560" alt="Screenshot 2023-01-19 at 11 42 07 PM" src="https://user-images.githubusercontent.com/5423313/213618788-7bf33774-e182-40c1-8bbf-ca5c149acd66.png">

Reply:
Ah interesting, I guess you could change that to `tokenDecimals!` instead. Will defer to whichever you prefer, feel free to resolve this comment yourself.